packages/webapp/src/components/collection/deckConstructionPanel.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react'
+import { Deck, Card } from 'src/store/types' 
+import Image from 'next/image' 
+
+interface DeckConstructionPanelProps {
+    deck: Deck
+    selectedCards: Card[]
+    onCardSelect: (card: Card) => void
+    onSave: (deck: Deck) => void
+    onCancel: () => void
+  }
+  
+  
+  const DeckConstructionPanel : React.FC<DeckConstructionPanelProps> = ({ deck, selectedCards = [], onCardSelect, onSave, onCancel }) => {
+    const [ deckName, setDeckName ] = useState(deck.name)
+    const [ isDeckNameValid, setIsDeckNameValid ] = useState(true)
+
+    const handleDeckNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setDeckName(event.target.value)
+      setIsDeckNameValid(true)

```

Comment:
We should probably check if the trimmed value is valid. Otherwise we can type only spaces and it seems like the deck name is valid.